pandas/formats/format.py

```diff
@@ -1951,7 +1951,11 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    values = levels.take(labels,
+                                         allow_fill=levels._can_hold_na,
```

Comment:
@Jeff that value is by default True, and once you pass a `fill_value`, you get an error for integer levels. So it is either passing `allow_fill` like above, or eiter passing `fill_value` conditionally like it was initially in this PR